### ISSUE TITLE
Filler not filling in water blocks

### ISSUE THREAD

**Issue by jburkej1** [BuildCraft/BuildCraft#4058]
The filler doesn't fill in water blocks. I have tried it with and without excavate turned on. 
The block flow seem much better than .14 alpha. 
thanks
j 

BuildCraft version: 7.99.15
Forge version: 8.0.99.99
Link to crash report or log: 
**Singleplayer** or multiplayer: 
Steps to reproduce: 
Additional information:

**Comment by LuigiHutch** [comment]
That is correct. Use the filler to build a box where you want the water to go and use a floodgate to fill the box with water.

**Comment by AEnterprise** [comment]
do you mean it doesn't replace water blocks with solid blocks?

**Comment by jburkej1** [comment]
it doesn't replace the water blocks. 
jb

**Comment by AEnterprise** [comment]
not sure if this should be default, excavate or another mode but it sure would be usefull

**Comment by jburkej1** [comment]
agreed. I usually don't fill in water holes before pit mining or worry about water sources found during mining as I am going to back fill it with the trash from another quarry anyway. Earlier versions had no issues filling in water blocks. I also noticed when I tested it at a different location that it wasn't removing grass and flowers as well.

**Comment by LuigiHutch** [comment]
Is there any practical uses for it becoming a separate mode? Otherwise this is just a bug and a regression in functionality.

**Comment by jburkej1** [comment]
not being a programmer but I would say it's just a bug as it worked fine in MC 1.10.x

**Comment by AEnterprise** [comment]
there is no official 1.10.x release, so whatever you used or are using on 1.10 it's not a good version to compare with for what should work how

**Comment by jburkej1** [comment]
I don't have the save game anymore maybe it was back with MC 1.8.9.

**Comment by jburkej1** [comment]
I have to say I really like the visual with the filler is set to clear, big multicolored spider destroying blocks like mad. nice work

**Comment by jburkej1** [comment]
went back and started a 1.8.9 game and the filler does fill water block and clears grass and flowers. It seems I skipped right over 1.9, 1.10, 1.11. no wonder things look different

**Comment by AlexIIL** [comment]
The fillers definition of what is solid and what is not is currently limited to "is it air?" I'm not against changing it to require actual solid (collidable) blocks for solid, and air (only) for not solid.

**Comment by AEnterprise** [comment]
a seperate mode to consider fluids (or non clolidable blocks) as empty would be usefull imo

**Comment by jburkej1** [comment]
I found a work around for small areas of water, I created a filler grid above the water. used a diamond sorter to send only sand and gravel to the filler. it kept right on pushing blocks out until the grid was filled. Took a while but it worked. Probably not a solution for large areas of water.

**Comment by AlexIIL** [comment]
Yeah, no-one has looked at this yet. Hopefully when some-one does (probably either me or afdw) we'll comment on this issue or link it to a github comment to indicate that it's been fixed - so you don't need to retest this bug in every newer version, as you can safely assume it hasn't been fixed yet.

**Comment by jburkej1** [comment]
Translation: go away and leave us alone :-)  
thanks for the update. 
Beta testing software for a school division is part of my job, I will be happy to test anything you need tested.
